File: app/src/main/java/com/dmi/perfectreader/layout/wordbreak/WordBreaker.java
package com.dmi.perfectreader.layout.wordbreak;

import com.dmi.util.annotation.Reusable;

import java.util.Locale;

public interface WordBreaker {
    WordBreaks breakWord(CharSequence text, Locale locale, int beginIndex, int endIndex);

    @Reusable
    interface WordBreaks {
        boolean canBreakBefore(int index);
    }
}

File: app/src/main/kotlin/com/dmi/perfectreader/layout/wordbreak/WordBreaker.kt
package com.dmi.perfectreader.layout.wordbreak

import com.dmi.util.annotation.Reusable
import java.util.*

interface WordBreaker {
    fun breakWord(text: CharSequence, locale: Locale, beginIndex: Int, endIndex: Int): WordBreaks

    @Reusable
    interface WordBreaks {
        fun canBreakBefore(index: Int): Boolean
    }
}
